app-tracker/app/repositories/page.tsx

```diff
@@ -1,10 +1,45 @@
-"use client"
-import React from 'react'
+"use client";
+import { useEffect, useState } from "react";
+import "./repositories.css";
+import "../spinner.css";
 
 function Repositories() {
-  return (
-    <div>Repositories</div>
-  )
+	const [repositories, setRepositories] = useState<any[]>([]);
+	const [loading, setLoading] = useState(true);
```

Comment:
I would move fetching and loading states to a custom hook returning the repositories and loading states

```
function Repositories() {
const {repositories, isLoading} = useCustomHook()
...
return (...)
}
function useCustomHook() {
...
return {repositories, isLoading}
}
```

This will help with refactoring later